src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/viewmodels/IndexViewModel.kt

```diff
@@ -1,15 +1,88 @@
 package org.vaccineimpact.orderlyweb.viewmodels
 
 import org.vaccineimpact.orderlyweb.ActionContext
-import org.vaccineimpact.orderlyweb.models.Report
+import org.vaccineimpact.orderlyweb.controllers.web.Serialise
+import org.vaccineimpact.orderlyweb.models.ReportVersion
+import org.vaccineimpact.orderlyweb.models.Scope
+import org.vaccineimpact.orderlyweb.models.permissions.ReifiedPermission
+import java.time.LocalDateTime
+import java.time.ZoneId
+import java.time.format.DateTimeFormatter
 
-data class IndexViewModel(val reports: List<Report>, val appViewModel: AppViewModel) : AppViewModel by appViewModel
+data class IndexViewModel(@Serialise("reportsJson") val reports: List<ReportRowViewModel>, val isReviewer: Boolean,
+                          val appViewModel: AppViewModel)
+    : AppViewModel by appViewModel
 {
-    constructor(context: ActionContext, reports: List<Report>)
-            : this(reports, DefaultViewModel(context, IndexViewModel.breadcrumb))
+    constructor(context: ActionContext, reports: List<ReportRowViewModel>, isReviewer: Boolean)
+            : this(reports, isReviewer, DefaultViewModel(context, breadcrumb))
 
     companion object
     {
         val breadcrumb = Breadcrumb("Main menu", "/")
+
+        fun build(reports: List<ReportVersion>, context: ActionContext): IndexViewModel
+        {
+            var currentKey = 0
+            val reportRows = reports.groupBy { it.name }.flatMap {
+                currentKey += 1
+                val parent = ReportRowViewModel.buildParent(currentKey, it.value)
+
+                val children = it.value.map { version ->
+                    currentKey += 1
+                    ReportRowViewModel.buildVersion(version, currentKey, parent)
+                }
+
+                children + parent
+            }
+
+            return IndexViewModel(context, reportRows,
+                    context.hasPermission(ReifiedPermission("reports.review", Scope.Global())))
+        }
+    }
+}
+
+data class ReportRowViewModel(val ttKey: Int,
+                              val ttParent: Int,
```

Comment:
`ttKey` and `ttParent` here will be serialised to `tt_key` and `tt_parent` in the json, as per the tree-table spec: 
https://github.com/reside-ic/tree-table/tree/prototype#basic-usage